great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+                if colname not in self._exclude_columns
+            ]
+
+        return table_column_names
+
+    def _generate_metric_configurations_to_check_cardinality(
+        self,
+        batch_ids: List[str],
+        column_names: List[str],
+    ) -> List[Dict[str, List[MetricConfiguration]]]:
+        """Generate metric configurations used to compute metrics for checking cardinality.
+
+        Args:
+            batch_ids: List of batch_ids used to create metric configurations.
+            column_names: List of column names used to create metric configurations.
+
+        Returns:
+            List of dicts of the form [{column_name: List[MetricConfiguration]},...]
+        """
+
+        limit_mode: CardinalityLimitMode = self._cardinality_checker.limit_mode
+
+        metric_name: str = limit_mode.related_metric_name
+
+        metric_configurations: List[Dict[str, List[MetricConfiguration]]] = []
+
+        for column_name in column_names:
+            metric_configurations.append(
+                {
+                    column_name: [
+                        MetricConfiguration(
+                            metric_name=metric_name,
+                            metric_domain_kwargs={
+                                "batch_id": batch_id,
+                                "column": column_name,
+                            },
+                            metric_value_kwargs=None,
+                            metric_dependencies=None,
+                        )
+                        for batch_id in batch_ids
+                    ]
+                }
+            )
+
+        return metric_configurations
+
+    def _columns_meeting_cardinality_limit(
+        self,
+        validator: Validator,
+        table_column_names: List[str],
+        metrics_for_cardinality_check: List[Dict[str, List[MetricConfiguration]]],
+    ) -> List[str]:
+        """Compute cardinality and return column names meeting cardinality limit.
+
+        Args:
+            validator: Validator used to compute column cardinality.
+            table_column_names: column names to verify cardinality.
+            metrics_for_cardinality_check: metric configurations used to
+                compute cardinality.
+
+        Returns:
+            List of column names meeting cardinality.
+        """
+        candidate_column_names: List[str] = table_column_names.copy()
+
+        # TODO AJB 20220218: Remove loops and refactor into single call to validator.get_metrics()
+        #  by first building up MetricConfigurations to compute. See GREAT-584
+        for metric_for_cardinality_check in metrics_for_cardinality_check:
+            for (
+                column_name,
+                metric_config_batch_list,
+            ) in metric_for_cardinality_check.items():
+                if column_name not in candidate_column_names:
+                    continue
+                else:
+                    for metric_config in metric_config_batch_list:
```

Comment:
@anthonyburdi It does not seem to me that `metric_config_batch_list` matches the actual type -- which is `MetricConfig`, I think -- so should it not be called `metric_config_list`?  Thanks.

Reply:
`metric_config_batch_list` is a list of MetricConfig so I think `metric_config` is named appropriately. `metric_config_batch_list` is so named because it is metric configs for each batch in batch_list. But open to suggestions for improvement.